scripts/ci/pr-bot/shared/persistentState.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const exec = require("@actions/exec");
+const fs = require("fs");
+const path = require("path");
+const { Pr } = require("./pr");
+const { ReviewersForLabel } = require("./reviewersForLabel");
+const { BOT_NAME } = require("./constants");
+
+export class PersistentState {
+  private switchedBranch: boolean;
```

Comment:
Initialize the variable here and drop the typing. That way you can remove the constructor.